oak/server/rust/oak_abi/src/lib.rs

```diff
@@ -16,11 +16,51 @@
 
 //! Type, constant and Wasm host function definitions for the Oak application binary interface.
 
+// TODO(#638): Generate from protobuf in a no_std compatible way
+#[cfg(feature = "std")]
 pub use proto::oak_api::ChannelReadStatus;
+#[cfg(feature = "std")]
 pub use proto::oak_api::OakStatus;
 
+#[cfg(feature = "std")]
 pub mod proto;
 
+// TODO(#638): Generate from protobuf in a no_std compatible way
+#[cfg(feature = "no_std")]
+pub mod proto {
```

Comment:
Would something like this work?

```rust
#[cfg(feature = "no_std")]
mod inner {
  pub enum ChannelReadStatus { /* ... */ }
  pub enum OakStatus { /* ... */ }
}

#[cfg(feature = "std")]
mod inner {
  mod proto;
  pub use proto::oak_api::{ChannelReadStatus, OakStatus};
}

pub use inner::*;
```

Reply:
I can't put the `mod proto;` inside `inner`, but I can do the rest, WDYT?

Reply:
LG! Are nested `mod`s not supported then?

Reply:
They are supported, but as that one in particular is a file import, rustc looks for the definition in `src/inner/proto` and using super/crate doesn't seem to work for this.